Source/WebCore/dom/TrustedType.cpp

```diff
@@ -302,12 +303,36 @@ ExceptionOr<String> requireTrustedTypesForPreNavigationCheckPasses(ScriptExecuti
         : nullString());
 }
 
-ExceptionOr<bool> canCompile(ScriptExecutionContext& scriptExecutionContext, JSC::CompilationType compilationType, String codeString, JSC::JSValue bodyArgument)
+ExceptionOr<bool> canCompile(ScriptExecutionContext& scriptExecutionContext, JSC::CompilationType compilationType, String codeString, const JSC::ArgList& args)
 {
-    if (bodyArgument.isObject())
-        return JSTrustedScript::toWrapped(scriptExecutionContext.vm(), bodyArgument) ? true : false;
+    if (compilationType == CompilationType::Function) {
```

Comment:
The spec says we should do that for DirectEval and IndirectEval. However, in that case this really simplifies to `if (args.at(0).isObject()) return true;`

See the MOZ_ASSERT I have in my Gecko patch: https://phabricator.services.mozilla.com/D229624

Isn't this covered by existing tests?

Reply:
This is covered in tests I'm pretty sure, it's just the model here can't directly match the spec flow for various reasons.